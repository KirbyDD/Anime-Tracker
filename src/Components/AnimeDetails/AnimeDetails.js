import React, {Component} from 'react';
import {connect} from 'react-redux';

class AnimeDetails extends Component {
  constructor(props){
    super(props);
  }

  addToWatchList = () => {
    
  }

  render(){
    return(
      <section>
        <img />
        <p>Status: </p>
        <h2>Title: </h2>
        <article>Synopsis
          <p></p>
        </article>
        <div>
          <p>Rating: </p>
          <p>Age Rating:</p>
        </div>
        <div>
          <p>Start Date</p>
          <p>End Date</p>
          <p>Number of Episodes</p>
          <button>Add To Watch List</button>
        </div>
      </section>
    )
  }
}

export default connect(null)(AnimeDetails);