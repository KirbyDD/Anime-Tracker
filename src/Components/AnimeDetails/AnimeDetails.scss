.anime-details {
  text-align: center;
  display: flex;
}

.details-styling {
  font-size: 20px;
  padding: 30px;
}
