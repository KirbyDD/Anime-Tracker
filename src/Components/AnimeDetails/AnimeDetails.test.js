import React from 'react';
import {shallow} from 'enzyme';
import {AnimeDetails} from './AnimeDetails';

let props = {
  id: 1,
  synopsis: 'This is the synopsis',
  enTitle: 'Life of TronKat',
  jpTitle: 'Tronkat no Life desu',
  episodes: 27,
  posterImage: {
    large: 'https://i.imgur.com/vihivug.jpg',
    tiny: null
  },
  coverImage: {
    small: 'https://i.imgur.com/vihivug.jpg',
    tiny: null
  },
  rating: '99.9',
  startDate: '1990-01-26',
  endDate: 'Still Playing',
  ageRating: '17+'
}

let props2 = {
  id: 3,
  enTitle: null,
  jpTitle: 'Tronkat no Life desu',
  episodes: 27,
  posterImage: {
    large: 'https://i.imgur.com/vihivug.jpg',
    tiny: null
  },
  coverImage: {
    small: 'https://i.imgur.com/vihivug.jpg',
    tiny: null
  },
  rating: '99.9',
  startDate: '1990-01-26',
  endDate: 'Still Playing',
  ageRating: '17+'
}

let result = {
  id: 1,
  synopsis: 'This is the synopsis',
  enTitle: 'Life of TronKat',
  jpTitle: 'Tronkat no Life desu',
  episodes: 27,
  posterImage: {
    large: 'https://i.imgur.com/vihivug.jpg',
    tiny: null
  },
  coverImage: {
    small: 'https://i.imgur.com/vihivug.jpg',
    tiny: null
  },
  currentEpisode: 1,
  comment: ''
}

let followedShows = [
  {
    id: 1,
    synopsis: 'This is the synopsis',
    enTitle: 'Life of TronKat',
    jpTitle: 'Tronkat no Life desu',
    episodes: 27,
    posterImage: {
      large: 'https://i.imgur.com/vihivug.jpg',
      tiny: null
    },
    coverImage: {
      small: 'https://i.imgur.com/vihivug.jpg',
      tiny: null
    },
    rating: '99.9',
    startDate: '1990-01-26',
    endDate: 'Still Playing',
    ageRating: '17+'
  }
]

describe('Anime Details', () => {
  let wrapper;
  let removeFromWatchList;
  let addToWatchList;

  beforeEach(function(){
    removeFromWatchList = jest.fn();
    addToWatchList = jest.fn();
    wrapper = shallow(
      <AnimeDetails {...props} followedShows={followedShows}
      removeFromWatchList={removeFromWatchList}
      addToWatchList={addToWatchList}/>
      )
    })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call checklist when button is clicked', () => {
    wrapper.instance().checkList = jest.fn();
    wrapper.find('button').simulate('click');
    expect(wrapper.instance().checkList).toHaveBeenCalled()
  })

  it('should call removeFromWatchList if show is found', () => {
    wrapper.instance().checkList(props.id);
    expect(removeFromWatchList).toHaveBeenCalled()
  })

  it('should call addToWatchList if show is found', () => {
    wrapper.instance().checkList(2);
    wrapper.instance().updateFavButton = jest.fn();
    expect(addToWatchList).toHaveBeenCalledWith(result)
  })

  it('should return Remove from Watch list', () => {
    expect(wrapper.instance().updateFavButton(1)).toEqual('Remove from Watch List')
  })

  it('should return Add to Watch list', () => {
    expect(wrapper.instance().updateFavButton(2)).toEqual('Add to Watch List')
  })

  it('should return English Title', () => {
    expect(wrapper.instance().titleToDisplay()).toEqual('Life of TronKat')
  })

  it('should return japanese Title', () => {
    wrapper = shallow(
      <AnimeDetails {...props2} followedShows={followedShows}
      removeFromWatchList={removeFromWatchList}
      addToWatchList={addToWatchList}/>
      )
    expect(wrapper.instance().titleToDisplay()).toEqual('Tronkat no Life desu')
  })
})