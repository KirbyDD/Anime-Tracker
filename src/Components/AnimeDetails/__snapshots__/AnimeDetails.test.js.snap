// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime Details should match snapshot 1`] = `
<section
  className="anime-details"
>
  <img
    className="poster"
    src="https://i.imgur.com/vihivug.jpg"
  />
  <div
    className="details-styling"
  >
    <h2>
      Title: 
      Life of TronKat
    </h2>
    <article>
      <h3>
        Synopsis
      </h3>
      <p>
        This is the synopsis
      </p>
    </article>
    <div>
      <p>
        Rating: 
        99.9
         
      </p>
      <p>
        Age Rating: 
        17+
      </p>
    </div>
    <div>
      <p>
        Start Date: 
        1990-01-26
      </p>
      <p>
        End Date: 
        Still Playing
      </p>
      <p>
        Number of Episodes: 
        27
      </p>
      <button
        onClick={[Function]}
      >
        Remove from Watch List
      </button>
    </div>
  </div>
</section>
`;
